tests/system/action/mediafile/test_create_directory.py

```diff
@@ -315,25 +328,139 @@ def test_create_directory_no_access_group_ids(self) -> None:
         assert model.get("is_directory") is True
         assert model.get("is_public") is True
 
-    def test_create_dictionary_no_permissions(self) -> None:
+    def test_create_directory_parent_id_parent_not_directory(self) -> None:
+        self.set_models(
+            {
+                "meeting/1": {
+                    "name": "meeting_1",
+                    "mediafile_ids": [7],
+                    "is_active_in_organization_id": 1,
+                },
+                "mediafile/7": {"owner_id": "meeting/1"},
+            }
+        )
+        response = self.request(
+            "mediafile.create_directory",
+            {
+                "owner_id": "meeting/1",
+                "title": "title_1",
+                "parent_id": 7,
+            },
+        )
+        self.assert_status_code(response, 400)
+        assert "Parent is not a directory." in response.json["message"]
+
+    def test_create_directory_parent_id_owner_mismatch(self) -> None:
+        self.set_models(
+            {
+                "meeting/1": {"mediafile_ids": [7], "is_active_in_organization_id": 1},
+                "meeting/2": {"is_active_in_organization_id": 1},
+                "mediafile/7": {"owner_id": "meeting/1", "is_directory": True},
+            }
+        )
+        response = self.request(
+            "mediafile.create_directory",
+            {
+                "owner_id": "meeting/2",
+                "title": "title_1",
+                "parent_id": 7,
+            },
+        )
+        self.assert_status_code(response, 400)
+        assert "Owner and parent don't match." in response.json["message"]
+
+    def test_create_directory_title_parent_id_unique(self) -> None:
+        self.set_models(
+            {
+                "meeting/1": {
+                    "mediafile_ids": [6, 7],
+                    "is_active_in_organization_id": 1,
+                },
+                "mediafile/6": {
+                    "is_directory": True,
+                    "owner_id": "meeting/1",
+                    "child_ids": [7],
+                },
+                "mediafile/7": {
+                    "title": "title_1",
+                    "owner_id": "meeting/1",
+                    "parent_id": 6,
+                },
+            }
+        )
+        response = self.request(
+            "mediafile.create_directory",
+            {
+                "owner_id": "meeting/1",
+                "title": "title_1",
+                "parent_id": 6,
+            },
+        )
+        self.assert_status_code(response, 400)
+        assert (
+            "Title 'title_1' and parent_id '6' are not unique."
```

Comment:
Please change this error message, e.g. "File x already exists in folder y" (and fetch the name of the folder to replace y)

Reply:
Renamed into  f"File '{title}' already exists in folder '{parent_title}'."
And updated the tests.